### ISSUE TITLE
`luminance_derive::UniformInterface` does not work when solely using `luminance_front`

### ISSUE THREAD

**Issue by Imberflur** [phaazon/luminance-rs#530]
The error is the `backend` module being absent. This happens because I'm using `luminance_front` aliased to `luminance` (it seems to be implied that  `luminance_front` is for quick projects, and I just need to do a small bit of rendering for an example). But I'm new to `luminance` so I'm not sure if this is a intended usage or if I am supposed to also need to use `luminance` directly.
```rs
error[E0433]: failed to resolve: could not find `backend` in `luminance`
   --> examples/simple.rs:108:17
    |
108 | #[derive(Debug, luminance_derive::UniformInterface)]
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ could not find `backend` in `luminance`
```

**Comment by Imberflur** [comment]
For now, I'm just switching to using `luminance` directly for everything.

**Comment by phaazon** [comment]
Argh thanks for the catch.

**Comment by phaazon** [comment]
Yeah, `UniformInterface` should be available with `luminance` via the `"derive"` feature gate, or you need `luminance-derive`.

The goal of `luminance-front` is to provide the « statically dispatched » backend types, not necessarily provide a quick-to-setup project. You can perfectly use `luminance-front` for bigger projects (at least to me). The places where you don’t want `luminance-front` are when you want to have a dynamic switch of backends, for instance, or if you have specific backends that are not supported by `luminance-front`.

I would also recommend not to use `luminance-front` for libraries that are general enough that they could be used with any backend, not only the backends `luminance-front` supports. The idea behind `luminance-front` is to re-export all symbols that are polymorphic in the backend type, but without the type variable. So for instance, a `luminance` type such as:

```rust
struct Foo<Backend, A, B, …>
```

becomes, in `luminance-front`

```rust
struct Foo<A, B, …>
```

And `Backend` is set to a specific value depending on how you compile / link your project. That logic is handled for you in `luminance-front`.
